src/secp256k1.c

```diff
@@ -769,6 +802,142 @@ int secp256k1_ec_pubkey_combine(const secp256k1_context* ctx, secp256k1_pubkey *
     return 1;
 }
 
+/* Compute an ec commitment tweak as hash(pubkey, data). */
+static int secp256k1_ec_commit_tweak(const secp256k1_context *ctx, unsigned char *tweak32, const secp256k1_pubkey *pubkey, const unsigned char *data, size_t data_size) {
```

Comment:
How about we let this take an initialized sha object. That way we can easily add domain separation via tagged hashes. And ideally we'd add a tagged hash for "ECDSA sign-to-contract" or something.